### DISCUSSION TITLE
customize login page

### DISCUSSION THREAD

**Discussion by gogi100** [rejetto/hfs#139] [General]
is exitisting a possible that i customize login page. i add a picture, some text?

**Comment by rejetto** [comment]
yes, but you need to know CSS. Do you?
have a look at https://github.com/rejetto/hfs/wiki/customization

**Reply by gogi100** [reply]
i set next code 
`<style>
.dialog-login form::before {
  content: "Dobro dosli na lokaciju DOKUMENTA, molim prijavite se"
}
</style>`
but login page does not changes
![1](https://user-images.githubusercontent.com/21057856/223502479-8683cff6-c090-4ef3-97ff-7c7e6ad9d0dd.jpg)

**Comment by gogi100** [comment]
i cannot find this section Admin-panel > Custom HTML

**Reply by rejetto** [reply]
what you did SHOULD work.
Custom header was replaced by the new "custom html" in the latest version.

**Reply by rejetto** [reply]
try updating to latest version, 0.38.2

**Reply by rejetto** [reply]
i just tried your code and it works here
<img width="478" alt="image" src="https://user-images.githubusercontent.com/1367199/223510774-1b5f5703-4d53-4dfe-8115-6637f17bece8.png">

**Reply by gogi100** [reply]
thanks

**Comment by gogi100** [comment]
i tryed insert picture with next code

`<style> 
.dialog-login form::before 
    { 
        content:url(/image/logo-dri.gif);
        content: "Добро дошли на локацију за пребацивање документације Документа.";
    } 

</style>`

nothing

**Reply by rejetto** [reply]
did you update to 0.38.2 ?
I have no idea why it's not working for you.
If you want you can give me access to your frontend, so i can give a look.
You can send me your address privately if you prefer, to rejetto@gmail.com

**Comment by gogi100** [comment]
i need help for customize dialog page. i rounded places that i want changing. i opened developing in chrome
![upload](https://user-images.githubusercontent.com/21057856/224247065-59f502d3-8358-438a-9272-507e86e06b18.jpg)

i want next:

- around the buttons i want color to #336699; i dont know how override style="display: flex; gap: 0.8em; flex-direction: column; position: sticky; top: -4px; background: var(--bg);"
- i want translate word 'finished' on serbian
- upper left corner i want hide

**Reply by rejetto** [reply]
you want to hide that icon just for upload or for all dialogs?

**Comment by gogi100** [comment]
i tryed changing color arround the buttons with
 div.dialog-content > div {
  background-color: #336699;
 }     
but nothing

**Reply by gogi100** [reply]
how i make darken the background behind dialog window?

**Reply by rejetto** [reply]
with this you change both dialog and inner box

.dialog, .dialog-content>div>div { background: #369; }

**Reply by rejetto** [reply]
darker like this?
.dialog-backdrop { background: #000c }

**Reply by gogi100** [reply]
thanks for the next code
`darker like this?
.dialog-backdrop { background: #000c }`
but this code is not resolves the problem, again white space around buttons
![upload](https://user-images.githubusercontent.com/21057856/224307857-15d67b94-24ab-4cf0-afcf-7ae2d0a581b4.jpg)

`.dialog, .dialog-content>div>div { background: #369; }`

also how reslove problems with

- i want translate word 'finished' on serbian, picture from post before
- upper left corner i want hide

**Reply by rejetto** [reply]
you are right, the code should have been
.dialog, .dialog-content>div>div { background: #369 !important; }

the other code "backdrop" is not for solving this problem, but to have darker background OUTSIDE the dialog

for the "finished" you have to wait next version. I will publish it soon.

for upper-left you can try `.dialog-icon { display: none }`

**Comment by rejetto** [comment]
So, you are trying to get blue dialogs but leaving white main page, right?
if yes, I can make the system easier, having 2 different colors, while now it is same for both

**Reply by gogi100** [reply]
yes, i try to make dialog with color #336699;. if you can make system that i can type color for dialog.

**Comment by rejetto** [comment]
the "finished" is not translated in current version but will be in next release.
I'm adding these in the italian translation

        "upload_finished": "{n,plural, one{# riuscito} other{# riusciti}} ({size})",
        "upload_errors": "{n,plural, one{# fallito} other{# falliti}}",

i think you can edit serbian file here at https://github.com/rejetto/hfs/edit/main/langs/hfs-lang-sr.json

**Reply by rejetto** [reply]
ok, version[ 0.38.3 ](https://github.com/rejetto/hfs/releases/tag/v0.38.3)is published and with it you can translate this text.
if you can please make the changes at https://github.com/rejetto/hfs/edit/main/langs/hfs-lang-sr.json
so other serbian people have update too.
Otherwise you can send me your new file

**Reply by gogi100** [reply]
i updated the serbian language json.

**Reply by rejetto** [reply]
I didn't receive any change or any file.
If you want, you can zip the json and attach it here, or send it to a@rejetto.com

**Reply by gogi100** [reply]
i sended zip file with serbian langage

**Comment by rejetto** [comment]
if instead you are trying to make all #369 not just the dialogs, it's easier to go like this

    :root { --bg: #369; --text: #fff; }
    a { color: #fff; text-decoration: underline; }

(example added on wiki)

**Comment by gogi100** [comment]
one more question, Is there a possibility that i create a page, when I click on link on it, it opens a login dialog?

**Reply by rejetto** [reply]
a "page" ? an html file?

**Reply by gogi100** [reply]
yes

**Reply by gogi100** [reply]
html page

**Reply by gogi100** [reply]
i noticed, when i type in admin-panel/options, field Title ДРИ, this word converted in DRI

**Reply by rejetto** [reply]
i don't know what DRI is. Please open new discussion/issue for different things, otherwise it is hard to follow all

**Reply by gogi100** [reply]
DRI is text that i typed in field TITLE

**Reply by rejetto** [reply]
now i understand, sorry I don't know Cyrillic :)
I tried but see no problem
<img width="241" alt="image" src="https://user-images.githubusercontent.com/1367199/224342215-94d4508c-a31e-4ed2-a75f-339d77e957d4.png">

<img width="313" alt="image" src="https://user-images.githubusercontent.com/1367199/224342148-06692426-4f5c-4d4b-ba1f-e120c7ca81d7.png">

**Comment by rejetto** [comment]
so you want to see the file/page in the list, but to ask for login when clicked.
I need to support the case with "can see" permission but not "can download".
Not possible with current version.
Maybe there is a workaround, but it's better to do things right. It should not be hard.

**Reply by rejetto** [reply]
> so you want to see the file/page in the list, but to ask for login when clicked. I need to support the case with "can see" permission but not "can download". Not possible with current version. Maybe there is a workaround, but it's better to do things right. It should not be hard.

please confirm this is what you need

**Reply by gogi100** [reply]
Yes, i confirm

**Reply by rejetto** [reply]
ok, i will try to include this in next version, anyway i plan to have it soon

**Reply by gogi100** [reply]
I see you set permissions access to see before login, but i dont know how on my page set link or button on login dialog?

**Reply by rejetto** [reply]
sorry, i don't understand your request.
You set "see" but not "download" on your page, and it will ask login, and then show the page.
Isn't this what you need?

**Reply by gogi100** [reply]
I thinked
I need to support the case with "can see" permission but not "can download".

I need on my first site page, how i set link to dialog login?

**Reply by rejetto** [reply]
You don't need to "link" the login, it is displayed automatically when needed.
If you have permission to see a file but not to download, you see it, you click it, and then you get login dialog.

What should your users first see when they enter your website?

**Reply by gogi100** [reply]
welcome page is
![welcomepage](https://user-images.githubusercontent.com/21057856/224778333-0a8527a3-fb24-4bd1-9c66-ede928f98eef.jpg)

**Reply by rejetto** [reply]
so, user enters your website and sees this page.
What next? Step by step

**Reply by gogi100** [reply]
on before version hfs 2.2, user clicks on link 'Пријавите се' then its opens the login window where the user type usename and password

**Reply by rejetto** [reply]
and what should users see after they do login?

**Reply by gogi100** [reply]
open next page
![capitalpage](https://user-images.githubusercontent.com/21057856/224785960-6bbbb48f-64b4-476f-b8f0-7931e1f4adae.jpg)

**Reply by rejetto** [reply]
then
1. set permission of download of folder "deljeni" to the accounts you want
2. make link "Пријавите се" point to "deljeni"

**Reply by gogi100** [reply]
i have a problem. in admin-panel when i try add index.html in home, i receive next message
![error](https://user-images.githubusercontent.com/21057856/224787508-bdf81a2a-e2d9-4fad-a7dc-1919586bda18.jpg)

**Reply by gogi100** [reply]
> then
> 
> 1. set permission of download of folder "deljeni" to the accounts you want
> 2. make link "Пријавите се" point to "deljeni"

but i have many users which redirected in different folders, no just to 'deljeni'

**Reply by rejetto** [reply]
the "invalid parent" is a bug. Will be fixed.
To workaround it, BEFORE you click "add", click on the X
<img width="199" alt="image" src="https://user-images.githubusercontent.com/1367199/224789325-365ba364-5218-49ba-a843-67b18ed1df85.png">

**Reply by rejetto** [reply]
i think that if you link "deljeni" is ok, and when the user logs in then he is redirected on other folder

**Reply by gogi100** [reply]
> i think that if you link "deljeni" is ok, and when the user logs in then he is redirected on other folder

but if i have the user 'test1' which must redirect to folder 'deljeni1', how i set link on my first page if i have two users, 'test' goes to folder 'deljeni' and 'test2' goes to 'deljeni1'?

**Reply by rejetto** [reply]
just link one of the protected folders.
did you try to see if it works?

**Reply by gogi100** [reply]
i dont know you understand structure of my folders. the structure is like on picture
![structure](https://user-images.githubusercontent.com/21057856/224798741-c6d591e1-a3ab-48ef-b506-f33484ee826b.jpg)

**Reply by rejetto** [reply]
I'm not sure what you want to do, but I can try.
You want people to see your index.html as they access your website?
Then I suggest: 
1. create a "web" folder inside your hfs-windows0.39  folder
2. move index.html inside "web", so it is the only file in it
3. click home, set "download" to "anyone"
4. set source to the "web" folder on disk
5. save
6. you now should see a new option "serve index.html". Enable it and save again.

This way people should be able to see the page.
I see that you now have your index.html directly in the VFS. I don't think it is working, right? Because it's not supported that way. Next version will support it.

did you try what I suggested?
I repeat: use a simple link to one of the protected folders, and configure your accounts to redirect to different folders. It should work.

**Reply by gogi100** [reply]
i repeated  your steps
`create a "web" folder inside your hfs-windows0.39 folder
move index.html inside "web", so it is the only file in it
click home, set "download" to "anyone"
set source to the "web" folder on disk
save`
but i have a error

https://user-images.githubusercontent.com/21057856/224807298-6241f2f3-18cd-4c9e-b0fa-5572ca846b97.mov

**Reply by rejetto** [reply]
I think it's because of a bug. YOu can see the "undefined" in your link.
<img width="166" alt="image" src="https://user-images.githubusercontent.com/1367199/224808993-1b3f403f-8f0e-4c2c-a889-18c7af34b3d8.png">

Please wait a minute, I'm publishing a fixed version.

**Reply by rejetto** [reply]
https://github.com/rejetto/hfs/releases/tag/v0.40.0

**Reply by gogi100** [reply]
> I'm not sure what you want to do, but I can try. You want people to see your index.html as they access your website? Then I suggest:
> 
> 1. create a "web" folder inside your hfs-windows0.39  folder
> 2. move index.html inside "web", so it is the only file in it
> 3. click home, set "download" to "anyone"
> 4. set source to the "web" folder on disk
> 5. save
> 6. you now should see a new option "serve index.html". Enable it and save again.
> 
> This way people should be able to see the page. I see that you now have your index.html directly in the VFS. I don't think it is working, right? Because it's not supported that way. Next version will support it.
> 
> did you try what I suggested? I repeat: use a simple link to one of the protected folders, and configure your accounts to redirect to different folders. It should work.

yes, the version 0.40 works. now, its opens my index.html but i don't understand  on which one of the protected folders? i have if you see 2 vfs: sektor 1, sektor 2 and two deljeni and deljeni1.

**Reply by gogi100** [reply]
i resolved my problem, i created the login vfs folder, i set permission for 'login' folder. in my link on first page i set /login/. it's ok. but i noticed that when i set permission on folder 'who can download' - no one and who can see- test. i cannot see my folders and files with account 'test'. in previous version of the hfs i could see files and folders, but not downloading. i need this option, can you enable in next version the hfs

**Reply by gogi100** [reply]
one more problem, i hided all breadcrumb, but i want  hide all breadcrumb, but last breadcrum, i selected on picture. how i make it
my code is:
`#root > header >a.breadcrumb {
display:none !important;
}`
![breadcrumb](https://user-images.githubusercontent.com/21057856/224952261-8a7d0792-5c1c-4eaf-a318-217ad97ed6d6.jpg)

**Reply by rejetto** [reply]
.breadcrumb:not(:last-of-type) { display:none }

**Reply by gogi100** [reply]
thanks, for help. can i open disscusion about two new topics:

- permission 'who can download'
- translating my custom solution without settings changing in web browser

if you can help me i customized the hfs per my requirements.

**Reply by gogi100** [reply]
for permission who can download:
- no one
![whocanwodnload-no one](https://user-images.githubusercontent.com/21057856/224988142-3848707e-aab3-427e-b1db-1c7201acdd05.jpg)
-any account (login required), select account 'test1'
![whocanwodnload-candownload](https://user-images.githubusercontent.com/21057856/224988435-8cf36cad-b7a8-480b-b56d-f58349d71201.jpg)

i need option that user can see folder documents and folders in this folder ;deljeni1' but he cannot download them

**Reply by rejetto** [reply]
you want list deljeni1 to be seen by anyone?

**Reply by rejetto** [reply]
set download to anyone, but then set the "masks"  field

```
"*":
  can_read: "*"
```

this means: any file in this folder has this can_read set to "any account".
the "*" in can_read means "any account"

**Reply by gogi100** [reply]
no, i want that in this folder  'test1' user can see and upload documents but that he cannot download  documents. on previos version the hfs 2.3 i set no download and it's works.

**Reply by rejetto** [reply]
ok, then set download of folder to "test1"
but then masks as
```
"*":
  can_read: false
```

**Reply by gogi100** [reply]
this mask disabled access to folders, i cannot see documents in folders

**Reply by rejetto** [reply]
so you want test1 to see all documents in all folders, but not download?

**Reply by gogi100** [reply]
Yes, in all folders in 'deljeni1' folder

**Reply by gogi100** [reply]
when i set mask on folder deljeni1
"*":
  can_read: false
i receive next

https://user-images.githubusercontent.com/21057856/225110133-d8d490c0-9161-4ef4-96da-96aa91f778b5.mov

**Reply by rejetto** [reply]
I'm thinking about this.
I guess the only decent solution would be to have a separate permission for accessing the folders, while at the moment i'm using the "download".

**Reply by gogi100** [reply]
can you implementate this solution to next version the hfs? one more requirement, the option that it's block uploading files with specific extension.

**Reply by rejetto** [reply]
please, stop using this place for all the requests. It is hard to follow this page. For the upload extension click [open new issue](https://github.com/rejetto/hfs/issues/new/choose).

I think I can implement what you described, but I need to understand why you want people to look inside all folders if they cannot download? Why not ask for login when you enter first folder?

**Reply by gogi100** [reply]
> I think I can implement what you described, but I need to understand why you want people to look inside all folders if they cannot download? Why not ask for login when you enter first folder?

because, i have users that must go to folder 'deljeni1', they upload their files, but they are cannot see files the other users. the users must go to folder 'deljeni' because of the file content or file type.

**Reply by rejetto** [reply]
I'm working on it, but it's not quick and easy to make.
My solution is to have a separated permission for listing of folders.
I must be careful to avoid security problems.

**Reply by rejetto** [reply]
Ok, now in 0.41 you have the "list" permission, so you can configure it differently from the "download"

**Comment by gogi100** [comment]
Can you enable this case u next version of the hfs. When i type www.domain.com its opens custom page, then i click on link or button, its open dialog for login?

**Reply by rejetto** [reply]
sorry I don't understand what you mean.
is this different from what you asked before about the page with login?

**Reply by gogi100** [reply]
I just know that you will enable this case in nrxt version of the hfs

**Reply by rejetto** [reply]
so you confirm you are talking about the same feature you were asking at https://github.com/rejetto/hfs/discussions/139#discussioncomment-5267405 ?

**Reply by gogi100** [reply]
Ok, i'm sorry

**Comment by gogi100** [comment]
one more question. i want set the customized hfs on internet. my server has public ip address, how i hide admin panel from access public. for example if some user types https://www.domain.com/~/admin it's opens admin panel login page. how i disable this option?

**Reply by rejetto** [reply]
at the moment there's no such option

**Comment by gogi100** [comment]
Many thanks

**Comment by gogi100** [comment]
Many thanks

**Comment by xrivo1** [comment]
In HFS 2.4.0, I inserted a string generated by a website, which shows the data of users who lynch. This is the result. With version 0.42.3, however, I failed.
![Login page](https://user-images.githubusercontent.com/106228981/229212591-b8d43b4c-297f-49e1-bf03-b65ec4055eb1.jpg)

**Reply by rejetto** [reply]
ti scrivo anche qui sennò forse non  ti arriva la notifica, ma ti sto rispondendo altrove: #186
